### PR TITLE
Make Aeternity work on Ubuntu 24

### PR DESCRIPTION
**Pull request by richcarl**
Uses the latest version of mnesia_rocksdb, which switches to the EMQX-maintained fork of the erlang rocksdb binding.

Libtool it's not automatic in recent Ubuntu releases and has to be listed explicitly in the build dependencies.

### CODE REVIEW

**Comment by ThomasArts** [comment]
I agree with @Hans that it would be nice for this specific commit to see the test results of both a complete sync and an upgrade. It seems so harmless to just use another database fork, but it may have rather severe consequences.
Could you run those tests and put the results as part of the PR?
There is no "issue" with the actual PR, but it needs this extra testing.

**Comment by richcarl** [comment]
So far it's just been proved to Work For Me (TM), running and syncing on my Ubuntu 24 machine. I agree that it should be tested a bit more before merging. I'll try to run it on something else (sadly my Mac is dead) so I can do an upgrade test.

**Comment by dincho** [comment]
I guess it still supports 20.04?

File: docs/build.md
Comment (dincho) [inline docs/build.md:25]: please keep the minimum supported version here (20.04)

**Comment by dincho** [comment]
> Looks good, but has it been tested properly? We need to sync Mainnet + Testnet with the new DB at least, and it would also be good to "upgrade" a node with the new DB - to check that there are no compatibility issues moving from old to new. I don't really expect any issues, but I think we should be a bit careful? Opinions @ThomasArts @dincho?

No idea what's the actual changes, can you please post a changelog? What's the difference with the old lib?

**Comment by richcarl** [comment]
> No idea what's the actual changes, can you please post a changelog? What's the difference with the old lib?

The old Erlang rocksdb library (the C level bindings) don't even compile on newer GCC version, and the maintainer (Benoit) seems inactive. The EMQX folks are very heavy users of Erlang rocksdb and have de facto taken over maintenance. See https://github.com/emqx/erlang-rocksdb/commits/master/ for the commit history for the last two years.

This PR pulls in the latest mnesia_rocksdb library (glue code for Mnesia to use rocksdb), which in its turn has been updated to use the new bindings.

**Comment by hanssv** [comment]
```
12:12:05.320 [info] Synced 1002695 generations: 86.75 ms/gen (33869128 mblock(s), 66918917 tx(s)) - Walltime: 90076.88s CPU: 86984.17s efficiency 96.57%
```
Mainnet ☝️

**Comment by hanssv** [comment]
```
08:53:50.712 [info] Synced blocks 1015863 - 1015930   (68 generations (28.91 ms/gen), 1092 mblock(s), 1153 tx(s))
08:53:51.285 [info] Removing/ending sync task #Ref<0.771012539.1461714947.89968> target was #{hash => <<255,125,224,95,189,116,141,132,234,198,218,96,35,109,7,49,46,61,177,176,0,183,241,225,205,229,138,181,190,175,14,121>>,height => 1015930}
08:53:51.285 [info] Synced 63441 generations: 34.13 ms/gen (1086901 mblock(s), 1188427 tx(s)) - Walltime: 2180.74s CPU: 2165.26s efficiency 99.29%
```
Testnet ☝️

**Comment by richcarl** [comment]
Turns out that the EMQX guys enabled LZ4 and Snappy by default, which makes it work for us, but at the same time they dropped the general flag passing variable that we were using to enable using a dynamically linked librocksdb in our docker build. That would make our build times go up again.

**Comment by dincho** [comment]
> Turns out that the EMQX guys enabled LZ4 and Snappy by default, which makes it work for us, but at the same time they dropped the general flag passing variable that we were using to enable using a dynamically linked librocksdb in our docker build. That would make our build times go up again.

Okay, so we keep using the system rocksdb (from the builder image) - both on CI and for tarballs. Still this PR switches to the EMQX fork to add support for ubuntu 24, right?

**Comment by richcarl** [comment]
I think this one is done now. Rebased and pointing to the fixed EMQX erlang-rocksdb, so that it can be built with a dynamic librocksdb, and works on Ubuntu 24 (see https://github.com/aeternity/aeternity/pull/4438). This means that in the containers, it will use RocksDB 7 just as before, except in the Ubuntu 24 container where it must use RocksDB 8 (see https://github.com/aeternity/docker-builder/pull/55).

**Comment by dincho** [comment]
If this change is final, please test full sync both on testnet and mainnet

**Review by dincho — APPROVED** [review]

**Review by hanssv — APPROVED** [review]

**Comment by richcarl** [comment]
```
10:11:13.456 [info] Synced blocks 1029822 - 1029874   (53 generations (102.67 ms/gen), 948 mblock(s), 1006 tx(s))
10:11:15.068 [info] Removing/ending sync task #Ref<0.1600434479.1901854721.173200> target was #{hash => <<227,87,65,3,99,24,13,236,243,54,241,84,233,42,242,215,107,90,146,175,0,111,175,154,61,10,173,18,79,132,162,70>>,height => 1029874}
10:11:15.068 [info] Synced 474817 generations: 132.19 ms/gen (11809188 mblock(s), 14879351 tx(s)) - Walltime: 62808.40s CPU: 62768.27s efficiency 99.94%
10:11:15.068 [info] Sync done (according to "pp_27xmgQ4...jxtFuAu")
```
Testnet ^

**Comment by richcarl** [comment]
```
10:11:13.456 [info] Synced blocks 1029822 - 1029874   (53 generations (102.67 ms/gen), 948 mblock(s), 1006 tx(s))
10:11:15.068 [info] Removing/ending sync task #Ref<0.1600434479.1901854721.173200> target was #{hash => <<227,87,65,3,99,24,13,236,243,54,241,84,233,42,242,215,107,90,146,175,0,111,175,154,61,10,173,18,79,132,162,70>>,height => 1029874}
10:11:15.068 [info] Synced 474817 generations: 132.19 ms/gen (11809188 mblock(s), 14879351 tx(s)) - Walltime: 62808.40s CPU: 62768.27s efficiency 99.94%
10:11:15.068 [info] Sync done (according to "pp_27xmgQ4...jxtFuAu")
```
Testnet^

**Pull request merged**
